### PR TITLE
Infer compression if file extension is uppercase

### PR DESCRIPTION
**Pull request by willbowditch**
Inferring compression fails for files with uppercase extensions (e.g. `x.zip` works but `y.ZIP` does not)

- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

**Comment by WillAyd** [comment]
Do we do this elsewhere? Outside of Windows most file systems I think are case sensitive so while relatively harmless I also am not sure this is worth doing

**Comment by willbowditch** [comment]
I think the path is retained with the original case and couldn't see any uses beyond identifying the compression method. Ran into this issue on OS X, tested locally with `read_csv` and capitalised zip files

**Comment by gfyoung** [comment]
> Do we do this elsewhere?

Not that I'm aware of.

> Outside of Windows most file systems I think are case sensitive so while relatively harmless I also am not sure this is worth doing

I *think* it would be okay to add this, but only after checking if we are on Windows.

cc @jreback

File: doc/source/whatsnew/v1.1.0.rst [hunk 1]
```diff
@@ -1050,6 +1050,7 @@ I/O
 - Bug in :meth:`~HDFStore.create_table` now raises an error when `column` argument was not specified in `data_columns` on input (:issue:`28156`)
 - :meth:`read_json` now could read line-delimited json file from a file url while `lines` and `chunksize` are set.
 - Bug in :meth:`DataFrame.to_sql` when reading DataFrames with ``-np.inf`` entries with MySQL now has a more explicit ``ValueError`` (:issue:`34431`)
+- Bug in :meth:`io.common.infer_compression` where capitalised files extensions were not decompressed by read_* functions.
```
Comment (jreback) [inline doc/source/whatsnew/v1.1.0.rst:1053]: can you add this PR number here (use the issue format). also add 'on windows platforms'

**Review by jreback — CHANGES_REQUESTED** [review]
i think this is ok to do everywhere.

**Comment by WillAyd** [comment]
Does pathlib offer any abstractions here? I don’t think it’s worthwhile for pandas to implement its own rules for file name equivalency.

File: doc/source/whatsnew/v1.1.0.rst (hunk 1 above)
Comment (willbowditch) [reply doc/source/whatsnew/v1.1.0.rst]: @jreback changed in a3558fd

**Comment by jreback** [comment]
this is fine; we already have our own rules for when to decompress - this is just expanding to capitalized which is not a big deal

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @willbowditch

**Pull request merged**